package.json

```diff
@@ -194,6 +203,10 @@
               "token": {
                 "description": "The Codecov API token for this endpoint (required for private repositories).",
                 "type": "string"
+              },
+              "type": {
+                "description": "Define hosted version control type (gh|ghe|bb|gl):",
```

Comment:
If these are the only values accepted you can add an `enum` field and list them there

Reply:
You can still add it in the JSON schema, which gives users validation hints when editing Sourcegraph settings :)

In TypeScript you can cast the value (but you might want to validate it first).